rust/azure_iot_operations_connector/src/data_processor/derived_json.rs

```diff
@@ -27,82 +23,43 @@ pub struct TransformError {
 #[derive(Debug, thiserror::Error)]
 enum TransformErrorRepr {
     #[error(transparent)]
-    JmespathError(#[from] JmespathError),
+    Jmespath(#[from] JmespathError),
     #[error(transparent)]
-    SerdeError(#[from] serde_json::Error),
+    Serde(#[from] serde_json::Error),
     #[error(transparent)]
-    SchemaError(#[from] MessageSchemaBuilderError),
-    #[error("Datapoints mapped multiple values to the same output field: {0}")]
-    DuplicateField(String),
+    Schema(#[from] MessageSchemaBuilderError),
 }
 
-/// Transform the input [`Data`] according to the JSON transformation defined in the [`Dataset`].
-/// Returns the transformed [`Data`] and a new [`MessageSchema`] that describes it.
+/// Returns a new [`MessageSchema`] that describes it.
 ///
 /// # Limitations
 /// - Cannot correctly interpret enums as it derives the schema only from JSON payload provided.
 /// - Similarly, optionality of fields cannot be inferred correctly in the schema.
-/// - Fields not present on the input JSON will be set to `null` in the output JSON if a transform
-///   for them is defined in the dataset.
-/// - Fields that are set to `null` as mentioned above will be set to `true` in the schema, as no
+/// - Fields that are set to `null` in the input JSON will be set to `true` in the schema, as no
 ///   information is available to derive the type of the field.
 ///
 /// # Errors
 /// Returns a [`TransformError`] if there is an error during the transformation or schema generation.
-pub fn transform(
-    mut data: Data,
-    dataset: &Dataset,
-) -> Result<(Data, MessageSchema), TransformError> {
+pub fn transform(data: Data) -> Result<MessageSchema, TransformError> {
     // NOTE: We delegate to a function here that modifies the data in place so that the entire
     // `data` struct does not need to be reallocated, while also being able to return it as part
     // of an error if necessary.
-    match transform_in_place_and_create_output_schema(&mut data, dataset) {
-        Ok(message_schema) => Ok((data, message_schema)),
+    match create_output_schema(&data) {
+        Ok(message_schema) => Ok(message_schema),
         Err(e) => Err(TransformError {
             repr: e,
             data: Box::new(data),
```

Comment:
similar, if we aren't taking Data, then we don't need to return it on the errors, since the customer will still have it